contracts/BalanceTrackerBase.sol

```diff
@@ -73,9 +73,48 @@ abstract contract BalanceTrackerFixedPriceBase {
         buyBackBurner = _buyBackBurner;
     }
 
+    /// @dev Adjusts initial requester balance accounting for max request delivery rate (debit).
+    /// @param balance Initial requester balance.
+    /// @param maxDeliveryRate Max delivery rate.
+    function _adjustInitialBalance(
+        address requester,
+        uint256 balance,
+        uint256 maxDeliveryRate,
+        bytes memory
+    ) internal virtual returns (uint256) {
+        // Check the request delivery rate for a fixed price
+        if (balance < maxDeliveryRate) {
+            // Get balance difference
+            uint256 balanceDiff = maxDeliveryRate - balance;
+            // Adjust balance
+            balance += _getRequiredFunds(requester, balanceDiff);
+        }
+
+        if (balance < maxDeliveryRate) {
+            revert InsufficientBalance(balance, maxDeliveryRate);
+        }
+
+        // Adjust account balance
+        return (balance - maxDeliveryRate);
+    }
+
+    /// @dev Adjusts final requester balance accounting for possible delivery rate difference (debit).
+    /// @param requester Requester address.
+    /// @param rateDiff Delivery rate difference.
+    /// @return Adjusted balance.
+    function _adjustFinalBalance(address requester, uint256 rateDiff) internal virtual returns (uint256) {
+        return mapRequesterBalances[requester] + rateDiff;
+    }
+
     /// @dev Drains specified amount.
+    /// @param amount Amount value.
     function _drain(uint256 amount) internal virtual;
 
+    /// @dev Gets fee composed of marketplace fee and another one, if applicable.
+    function _getFee() internal view virtual returns (uint256) {
+        return IMechMarketplace(mechMarketplace).fee();
+    }
```

Comment:
This allows us to track additional fee if it's coming from other sources, like NVM

Reply:
So the 1% NVM fee?